src/goto-programs/goto_trace.cpp

```diff
@@ -75,7 +76,10 @@ void goto_trace_stept::output(
   case goto_trace_stept::SHARED_WRITE: out << "SHARED WRITE"; break;
   case goto_trace_stept::FUNCTION_CALL: out << "FUNCTION CALL"; break;
   case goto_trace_stept::FUNCTION_RETURN: out << "FUNCTION RETURN"; break;
-  default: assert(false);
+  default: {
+    out << "unknown type: " << type << std::endl;
+    assert(false);
+    }
```

Comment:
I'm generally in favour of omitting the "default" clause when all cases are covered, so as to let the compiler warn about this. But that view isn't necessarily shared by others.

That said: the placement of { and } does not match coding guidelines.